react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,66 @@
+import { useState, useEffect, useRef, useCallback } from 'react';
+import NextButton from './NextButton';
+import PrevButton from './PrevButton';
+import Dots from './Dots';
+import ImageCard from './ImageCard';
+
+type Image = {
+  src: string;
+  alt: string;
+};
+
+type Props = {
+  jjk: Image[];
+};
+
+export function Carousel({ jjk }: Props) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const intervalRef = useRef<number | null>(null);
```

Comment:
Congrats on figuring out how to use `useRef`!

The instructor solution I showed you is able to accomplish the same thing using the useEffect cleanup function and the properties of closures. But I'm impressed with how you've solved this exercise.